### ISSUE TITLE
Make tick counts better for the day time units

### ISSUE THREAD

**Issue by willium** [vega/vega-lite#6593]
Forked from #6589
This issue is specifically addressing wacky axes created in Vega-Lite for non-continuous timeUnits

Examples (click images to see spec in editor):
[![image](https://user-images.githubusercontent.com/356976/83956274-0cbf5e80-a811-11ea-9386-efed0dec4b77.png)](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEySYUqUAwBOgtCrVICUJNohSZ8gL5Lk2gNb6mpxSEzaksiADM22hGgA2qDeNIKYcNr6qgCeABSqmAwIlACacKYAlAAEAKTZAEwgVgC6SnCyUGzKNLJkaKAAHg0goXCCyvrpLkqYMThw+hEIOH5Ier00CHAAqrJ00UgxxUrLmiBIZGTafBH6VQyymK59A-oAjgyemHRqNKTFVlZAA)

[![image](https://user-images.githubusercontent.com/356976/83956284-36788580-a811-11ea-8601-3c452a92b626.png)](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAdxoATemhBQ5mJDVlwATopBqketKAZHBW83oJQkRiFJnyAvkuRGA1lqYNFABSphGSLIQAGZsRghoANqg0TSCmMb2SACeABQODAiUAJpwLgCUAAQApJUATCCeALpKcLI6agZkViAAHj2pcIJqWqUuppjZOHCBiDhxSHZhNAhwAKqydFnZjUo7qKCT01oAjgyRQXpBpKaDw1oAIjQQOIJIUIhtmI2enkA)

I believe that by using Vega's [timeSequence](https://vega.github.io/vega/docs/expressions/#timeSequence) expression, Vega-Lite can make better axis labels.

Vega example: 
[![image](https://user-images.githubusercontent.com/356976/83956297-527c2700-a811-11ea-8498-cf08ffa11e67.png)](https://vega.github.io/editor/#/examples/vega/time-units)

**Comment by kanitw** [comment]
So you're suggesting that instead of fixing this in Vega/D3, we should do it here?

**Comment by willium** [comment]
Yes, I believe Vega's timeSequence expression should make it possible to do from Vega-Lite.

**Comment by jwoLondon** [comment]
An alternative and possibly one that is more flexible would be to create a temporal sequence generator similar to the current [numeric sequence generator](https://vega.github.io/vega-lite/docs/data.html#sequence). 

A user could then create a spec to generate a temporal sequence of any desired resolution/extent and then perform a lookup transformation to combine with the temporal dataset that contains gaps, setting a failed lookup appropriately.

**Comment by kanitw** [comment]
@willium Feel free to take a stab. 

@jwoLondon That sounds like a good idea, but that would require modifying Vega.  If you would like to have that, please file an issue there.
